packages/web-components/src/components/va-table/va-table.tsx

```diff
@@ -40,157 +50,168 @@ export class VaTable {
-      );
-      button.innerHTML = `${columns[this.sortColumn]}${icon}`;
-      button.onclick = this.handleSort.bind(this);
-
-      const sortableHeader = this.headers[this.sortColumn];
-      sortableHeader.childNodes.forEach(child => child.remove());
-      sortableHeader.appendChild(button);
-
-      this.handleSort();
+  /**
+   * Generate an array of span elements that are inside 
+   * a particular va-table-row element
+   */
+  getCellsInRow(row: Element): HTMLSpanElement[] {
+    const children = row.querySelectorAll('span');
+    const cells = Array.from(children);
+    if (!this.cols) {
+      this.cols = cells.length;
     }
+    return cells;
+  }
 
-    // Store alignment classes by column index.
-    const alignment = {};
-
-    const handleCellAdjustments = () => {
-      const rows = Array.from(this.el.querySelectorAll('va-table-row'))
-        .filter((row) => row.slot !== 'headers') as Array<HTMLVaTableRowElement>;
-
-      Array.from(rows).forEach((row, index) => {
-        const cells = elementChildren(row);
-
-        cells.forEach((cell: HTMLSpanElement, colNum) => {
-          // Look at the first row of data to determine type of data in column
-          // Right align columns with numeric data
-          if (index === 0 && isNumeric(cell.textContent)) {
-            // eslint-disable-next-line i18next/no-literal-string
-            alignment[colNum] = 'text-align-right';
-            (this.headers[colNum] as HTMLElement).classList.add(
-              alignment[colNum],
-            );
-          }
-          if (alignment[colNum]) {
-            cell.classList.add(alignment[colNum]);
-          }
-  
-          // This allows the responsive table in mobile view to display
-          // a column header
-          /* eslint-disable i18next/no-literal-string */
-          cell.setAttribute('data-label', columns[colNum]);
-          cell.setAttribute('role', 'cell');
-        });
+  /**
+   * Generate an array holding the va-table-rows from the component's slot
+   */
+  getRows(): Element[] {
+    const rows = Array.from(this.el.querySelectorAll('va-table-row'));
+    if (!this.rows) {
+      this.rows = rows.length;
+    }
+    return rows;
+  }
+
+  /**
+   * Generate an array of all the spans in all the va-table-rows 
+   * in the slot in order starting with row 0, column 0.
+   */
+  getAllCells(): void {
+    const cells: HTMLSpanElement[] = [];
+    let count = 0;
+    for (const row of this.getRows()) {
+      const cellsInRow = this.getCellsInRow(row);
+      cellsInRow.forEach((cell) => {
+        cell.setAttribute('slot', `va-table-slot-${count}`);
+        count++;
       });
-    };
+     
+      cells.push(...cellsInRow);
+    }
+    this.cells = cells;
+  }
 
-    handleCellAdjustments();
+  /**
+   * Generate a DocumentFragment that holds the cells 
+   * to be slotted into a va-table-inner component
+   */
+  makeFragment(): DocumentFragment {
+    const frag = document.createDocumentFragment();
```

Comment:
This is really interesting. Is the purpose of using `DocumentFragment` instead of direct dom manipulation because of cost savings? (speed) Or is there a specific web component reason to use it?

Reply:
@jamigibbs - I didn't run any tests as far as speed but it seemed the easiest way to add the cells and from the research I did seemed the preferred way (but I could be wrong).